addons/web_editor/static/src/js/wysiwyg/widgets/link_dialog.js

```diff
@@ -159,6 +159,7 @@ var LinkDialog = Dialog.extend({
         this.data.className = this.data.iniClassName
             .replace(allBtnClassSuffixes, ' ')
             .replace(allBtnShapes, ' ');
+        this.isSubmitAnchor = this.data.className.indexOf('s_website_form_send') !== -1;
```

Comment:
You don't need a new variable for that. Just set the `isButton` value to `true` in such a case.

Reply:
I added this variable to make difference between a button element and an anchor handled as a button (e.g. submit anchor) since we replace the preview element (`<button/>` or `<a/>`) according to the target node type.

Reply:
Mmh, the preview element should look the same whether it is a button or a link anyway. So I would still simplify the code to not have to do `if (!this.isSubmitAnchor && !this.isButton` everywhere. The preview element style only depends on the classes, it can stay being a link just like before and it would be an even more stable fix this way.